tools/createRdFiles.R

```diff
@@ -78,56 +80,124 @@ usage<-function(name, args) {
+# any conversion of Sphinx text to Latex text goes here
+convertSphinxToLatex<-function(raw) {
+	changeSphinxHyperlinksToLatex(raw)
 }
 
-# doc has 
-#:param xyz:
-# :returns:
-# Example::
-# :: (end of parameters?)
-# TODO other formatting
-processDetails<-function(raw) {
-	# this replaces ':param <param name>:' with '<param name>:'
-	result<-gsub(":param (\\S+):", "\n\\1:", raw)
-	result<-gsub(":returns:", "returns:", result)
+changeSphinxHyperlinksToLatex<-function(raw) {
+	gsub("`([^<\n]*) <([^>\n]*)>`_", "\\\\href{\\2}{\\1}", raw)
+}
+
+parseArgDescriptionsFromDetails<-function(raw) {
+	# escape any escaped-escapes
+	preprocessed<-gsub("\\\\", "\\\\\\\\", raw)
+	# change all quotes to escaped quotes
+	preprocessed<-gsub("\"", "\\\\\"", preprocessed)
+	# change \r\n to \n
+	preprocessed<-gsub("\r\n", "\n", preprocessed)
+
+	# find parameters and convert them, along with their def'ns, to json
+	# reminder: \w in a regexp means "word character", [A-Za-z0-9_]
+	json<-gsub(":(parameter|param|type|var) (\\w+):", "\",\"\\2\":\"", preprocessed)
+	# prepend "{\"unusedPrefix\":\""
+	# add "\"}" to the end
+	json<-paste0("{\"unusedPrefix\":\"", json, "\"}")
+	# parse JSON into named list
+	paramsList<-fromJSON(json)
+	# truncate each entry at double-new-line
+	result<-lapply(paramsList, 
+		function(x) {
+			p<-regexpr("\n\n", x)
+			if(p<0)return(x)
+			substr(x,1,p-1)
+		}
+	)
+	result$unusedPrefix<-NULL
+	result
+}
+
+pyVerbiageToLatex<-function(raw) {
 	# TODO there might be convertable content AFTER the double colon.  One case is 
 	# that in which the :returns: field comes after the example.
-	result<-gsub("Example::.*$", "", result) # remove 'Example::' and everything following
+	result<-gsub("Example::.*$", "", raw) # remove 'Example::' and everything following
 	result<-gsub("::.*$", "", result) # remove '::' and everything following
+	# this replaces ':param <param name>:' with '\nparam name:'
+	# same for parameter, type, var
+	result<-gsub(":(parameter|param|type|var) (\\w+):", "\n\\2:", result)
+	result<-gsub(":returns:", "returns:", result)
+	result<-gsub(":py:class:`(\\S+\\.)*(\\S+)`", "\\2", result)
+	
+	convertToUpper<-"##convertToUpper##" # marks character to convert
+	result<-gsub(":py:mod:`(\\S+\\.)*(\\S+)`", paste0(convertToUpper,"\\2"), result)
+	result<-gsub(":py:func:`Synapse.(\\w+)`", paste0("syn", convertToUpper,"\\1"), result)
+	result<-gsub(":py:func:`synapseclient.Synapse.(\\w+)`", paste0("syn",convertToUpper,"\\1"), result)
+	result<-gsub(":py:meth:`synapseclient.Synapse.(\\w+)`", paste0("syn",convertToUpper,"\\1"), result)
+	while (TRUE) {
+		ctuIndex<-regexpr(convertToUpper, result)[[1]]
+		if (ctuIndex<0) break
+		lcChar<-nchar(convertToUpper)+ctuIndex
+		result<-paste0(substring(result,1,ctuIndex-1), 
+				toupper(substring(result,lcChar,lcChar)), 
+				substring(result, lcChar+1))
+	}
+
+	result<-gsub(dictDocString, "\nConstructor accepts arbitrary named arguments.\n", result, fixed=TRUE)
```

Comment:
Is this the right description of kwargs?  Maybe "Constructor accepts named arguments."